README.md
# Svelte-Sidebar-Menu

[![npm version](https://badgen.net/npm/v/svelte-sidebar-menu)](https://www.npmjs.com/package/svelte-sidebar-menu)
[![npm downloads](https://badgen.net/npm/dw/svelte-sidebar-menu)](https://www.npmjs.com/package/svelte-sidebar-menu)
[![release](https://badgen.net/github/release/shinokada/svelte-sidebar)](https://github.com/shinokada/svelte-sidebar/releases)
[![license](https://badgen.net/npm/license/svelte-sidebar-menu)](https://github.com/shinokada/svelte-sidebar/blob/main/LICENSE)

## Installation

```sh
npm i -D svelte-sidebar-menu@latest
```

## Demo

[Svelte-Sidebar-Menu](https://svelte-sidebar.vercel.app/)

## Features

- [Responsive](https://svelte-sidebar.vercel.app/responsive/top-fix)
- Click outside to close the sidebar.
- [Flowbite-Svelte sidebar integration](https://flowbite-svelte.com/sidebars)
- [Inert](https://svelte-sidebar.vercel.app/inert)
- Svelte trainstions, [fly-x](https://svelte-sidebar.vercel.app/transitions/fly-x), [fly-y](https://svelte-sidebar.vercel.app/transitions/fly-y), [fade](https://svelte-sidebar.vercel.app/transitions/fade),[slide](https://svelte-sidebar.vercel.app/transitions/slide), and [blur](https://svelte-sidebar.vercel.app/responsive-transition/blur).
- [No transition](https://svelte-sidebar.vercel.app/responsive-no-transition/top-fix)
- [Sidebar style](https://svelte-sidebar.vercel.app/sidebar-custom-style)
- [Sidebar style 2](https://svelte-sidebar.vercel.app/multi-custom-style)
- [Top menu](https://svelte-sidebar.vercel.app/sidebar-topmenu)
- [Fixed top menu](https://svelte-sidebar.vercel.app/fixed-menu)
- [Scroll sidebar](https://svelte-sidebar.vercel.app/scroll-sidebar)

## Inert

[HTMLElement.inert](https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement/inert) makes the browser to ignore user input events for the element. For example, when a user press a tab key, it cycles down links and input fields.

Test [Svelte-Sidebar demo](https://svelte-sidebar.vercel.app/inert) by pressing the tab keyboard when the sidebar closed and opened. When the sidebar is closed, the tab ignores the sidebar links and when it is open the tab goes throught the menu links.

## Responsive Top Menu Fixed

[Repo](https://github.com/shinokada/svelte-sidebar/tree/main/src/routes/responsive)

[The demo](https://svelte-sidebar.vercel.app/responsive/top-fix) set the breaking point at 1024. When the window is greater than 1024px, the sidebar is kept open. When the winddow is less than 1024px, the sidebar is open by the bars at top-left.

To make the sidebar responsive, add the following to all files:

## Responsive no transition

[Repo](https://github.com/shinokada/svelte-sidebar/tree/main/src/routes/responsive-no-transition)

## Flowbite-Svelte Sidebar

- [Demo](https://svelte-sidebar.vercel.app/flowbite-svelte-sidebar)
- [Repo](https://github.com/shinokada/svelte-sidebar/tree/main/src/routes/flowbite-svelte-sidebar)

## Transition

[Repo](https://github.com/shinokada/svelte-sidebar/tree/main/src/routes/transitions)

## Top menu fixed

[Repo](https://github.com/shinokada/svelte-sidebar/blob/main/src/routes/fixed-menu.svelte)

## Inert

[Repo](https://github.com/shinokada/svelte-sidebar/blob/main/src/routes/inert.svelte)

## Custome style

- [Repo 1](https://github.com/shinokada/svelte-sidebar/blob/main/src/routes/multi-custom-style.svelte)
- [Repo 2](https://github.com/shinokada/svelte-sidebar/blob/main/src/routes/sidebar-custom-style.svelte)

## Scroll

[Repo](https://github.com/shinokada/svelte-sidebar/blob/main/src/routes/scroll-sidebar.svelte)

## Top menu (not fixed)

[Repo](https://github.com/shinokada/svelte-sidebar/blob/main/src/routes/topmenu.svelte)

## Article

[A Svelte Sidebar Menu Component Implemented with Tailwind CSS](https://medium.com/mkdir-awesome/a-svelte-sidebar-menu-component-implemented-with-tailwind-css-c039b23010e)


CHANGELOG.md
# Changelog

All notable changes to this project will be documented in this file. See [standard-version](https://github.com/conventional-changelog/standard-version) for commit guidelines.

### [0.9.12](https://github.com/shinokada/svelte-sidebar/compare/v0.9.11...v0.9.12) (2022-10-07)


### Bug Fixes

* remove aria-controls=mobile-menu-2 ([1d8b58d](https://github.com/shinokada/svelte-sidebar/commit/1d8b58d2b6fbe8ba7c88cc23161d076e9869c5ca))

### [0.9.11](https://github.com/shinokada/svelte-sidebar/compare/v0.9.10...v0.9.11) (2022-08-30)

### [0.9.10](https://github.com/shinokada/svelte-sidebar/compare/v0.9.9...v0.9.10) (2022-08-30)

### [0.9.9](https://github.com/shinokada/svelte-sidebar/compare/v0.9.8...v0.9.9) (2022-08-30)


### Features

* add 3 columns ([90bd47f](https://github.com/shinokada/svelte-sidebar/commit/90bd47f75140025991be5197f8b6791503a22711))


### Bug Fixes

* add engines node version >=16.0.0 ([8c63be8](https://github.com/shinokada/svelte-sidebar/commit/8c63be8d8fadbc30055438d409f7a92796fbf399))
* update 3-columns page ([9410d5d](https://github.com/shinokada/svelte-sidebar/commit/9410d5d33e4d348e907e6d9378bcff3321193973))

### [0.9.8](https://github.com/shinokada/svelte-sidebar/compare/v0.9.7...v0.9.8) (2022-05-19)


### Bug Fixes

* update creatporps and props page ([642371b](https://github.com/shinokada/svelte-sidebar/commit/642371b3c6232a022450d22978601e64fe351be1))

### [0.9.7](https://github.com/shinokada/svelte-sidebar/compare/v0.9.6...v0.9.7) (2022-05-19)

### Bug Fixes

- add title to all pages ([9f36c49](https://github.com/shinokada/svelte-sidebar/commit/9f36c49710e54f53a08f7e262a480c7ce76d23c3))
- update creatporps and props page ([e529389](https://github.com/shinokada/svelte-sidebar/commit/e529389e23c185bb7c3133e957d0312429915629))
- update description ([2ea0efa](https://github.com/shinokada/svelte-sidebar/commit/2ea0efa5affa1450f3310948bccf78c61b1ecfe0))

### [0.9.6](https://github.com/shinokada/svelte-sidebar/compare/v0.9.5...v0.9.6) (2022-05-16)

### Bug Fixes

- add topDiv to Side TopMenu in the else statement ([1d9453a](https://github.com/shinokada/svelte-sidebar/commit/1d9453af5254f3b6de8c9cf4c96f16de82111aaa))

### [0.9.5](https://github.com/shinokada/svelte-sidebar/compare/v0.9.4...v0.9.5) (2022-05-16)

### Bug Fixes

- add topDiv to Side TopMenu ([f5c3de2](https://github.com/shinokada/svelte-sidebar/commit/f5c3de28fab184bb2005fd61e8d8abaf8d6ddfd4))

### [0.9.4](https://github.com/shinokada/svelte-sidebar/compare/v0.9.3...v0.9.4) (2022-05-16)

### Bug Fixes

- add topDiv to Side ([3078251](https://github.com/shinokada/svelte-sidebar/commit/30782514464c873d01cc738b56d007dcc1c0f320))

### [0.9.3](https://github.com/shinokada/svelte-sidebar/compare/v0.9.2...v0.9.3) (2022-05-16)

### Bug Fixes

- add pl-9 to flowbite-svelte-sidebar \_\_layout ([8fdca6c](https://github.com/shinokada/svelte-sidebar/commit/8fdca6c005db1f8a0f1d637e467934a2d74a2bc4))
- add topDiv to TopMenu ([c2fba33](https://github.com/shinokada/svelte-sidebar/commit/c2fba339a332a9f1bc8c5bb4e5c6d74ebe596450))
- update flowbite-svelte to 0.15.35 ([5491263](https://github.com/shinokada/svelte-sidebar/commit/5491263ee67bdcfa897e7081d799f6b105607a93))

### [0.9.2](https://github.com/shinokada/svelte-sidebar/compare/v0.9.1...v0.9.2) (2022-05-15)

### Bug Fixes

- add 'main': 'index.js', to package.json ([55e2a80](https://github.com/shinokada/svelte-sidebar/commit/55e2a800b2afcada91faef8fa4ca13c8441ffb08))

### [0.9.1](https://github.com/shinokada/svelte-sidebar/compare/v0.9.0...v0.9.1) (2022-05-15)

### Bug Fixes

- add class to tailwind.config.cjs and flowbite-svelte-sidebar test ([a046497](https://github.com/shinokada/svelte-sidebar/commit/a046497ad6373742974b9ad648559e3e9556607b))

## [0.9.0](https://github.com/shinokada/svelte-sidebar/compare/v0.8.11...v0.9.0) (2022-05-15)

### Features

- add flowbite-svelte sidebar ([f4692f7](https://github.com/shinokada/svelte-sidebar/commit/f4692f7a8394437fdb13417bf8cc61405d3fa4dc))

### [0.8.11](https://github.com/shinokada/svelte-sidebar/compare/v0.8.10...v0.8.11) (2022-05-14)

### Bug Fixes

- add lg:hidden to hamburger menu ([cf16dc6](https://github.com/shinokada/svelte-sidebar/commit/cf16dc67979a367c590a887f60a578a5010ecfe4))

### [0.8.10](https://github.com/shinokada/svelte-sidebar/compare/v0.8.9...v0.8.10) (2022-05-14)

### [0.8.9](https://github.com/shinokada/svelte-sidebar/compare/v0.8.8...v0.8.9) (2022-05-14)

### [0.8.8](https://github.com/shinokada/svelte-sidebar/compare/v0.8.7...v0.8.8) (2022-05-14)

### [0.8.7](https://github.com/shinokada/svelte-sidebar/compare/v0.8.6...v0.8.7) (2022-05-13)

### Bug Fixes

- add : string to exported props ([8f38866](https://github.com/shinokada/svelte-sidebar/commit/8f38866f5d9dbf23f237ae88f9b419351a630a03))

### [0.8.6](https://github.com/shinokada/svelte-sidebar/compare/v0.8.5...v0.8.6) (2022-05-09)

### Bug Fixes

- name change from store to sidebarStore.ts ([3929ff1](https://github.com/shinokada/svelte-sidebar/commit/3929ff12f69721a845e9f18d7cc4ec0b5510fb33))

### [0.8.5](https://github.com/shinokada/svelte-sidebar/compare/v0.8.4...v0.8.5) (2022-05-09)

### Bug Fixes

- add OutsideClick to index ([6e59988](https://github.com/shinokada/svelte-sidebar/commit/6e599882e2d396a840a4895005352d1ceb2c1514))

### [0.8.4](https://github.com/shinokada/svelte-sidebar/compare/v0.8.3...v0.8.4) (2022-05-09)

### [0.8.3](https://github.com/shinokada/svelte-sidebar/compare/v0.8.2...v0.8.3) (2022-05-09)

### [0.8.2](https://github.com/shinokada/svelte-sidebar/compare/v0.8.1...v0.8.2) (2022-05-09)

### [0.8.1](https://github.com/shinokada/svelte-sidebar/compare/v0.8.0...v0.8.1) (2022-05-08)

## [0.8.0](https://github.com/shinokada/svelte-sidebar/compare/v0.7.6...v0.8.0) (2022-05-08)

### Features

- add OutsideClick component ([be8af15](https://github.com/shinokada/svelte-sidebar/commit/be8af1575ad0c900d941e1bbe9dc300fa9901e38))

### Bug Fixes


src/routes/menus.ts
import type { MenuType, TopMenuType } from '$lib/types';

export const mainMenuList: MenuType[] = [
	{ href: '/responsive/top-fix', rel: 'external', name: 'Responsive' },
	{ href: '/responsive-no-transition/top-fix', rel: 'external', name: 'Responsive No Transition' },
	{ href: '/transitions/blur', rel: 'external', name: 'Transitions' },
	{ href: '/topmenu', rel: 'external', name: 'Other menus' },
	{ href: '/flowbite-svelte-sidebar', rel: 'external', name: 'Flowbite-Svelte Sidebar' },
	{ href: '/props', rel: 'external', name: 'Props' },
	{ href: 'https://www.npmjs.com/package/svelte-sidebar-menu', name: 'NPM' },
	{ href: 'https://github.com/shinokada/svelte-sidebar', name: 'GitHub' }
];

export const topMenuList: MenuType[] = [
	{ href: '/responsive/top-fix', rel: 'external', name: 'Responsive' },
	{ href: '/responsive-no-transition/top-fix', rel: 'external', name: 'Responsive No Transition' },
	{ href: '/transitions/blur', rel: 'external', name: 'Transitions' },
	{ href: '/three-columns', rel: 'external', name: '3 Columns' },
	{ href: '/topmenu', rel: 'external', name: 'Other menus' }
];

export const menuListResponsive: MenuType[] = [
	{ href: '/responsive/top-fix', name: 'Responsive Top Fixed' },
	{ href: '/responsive/top-fix-2', name: 'Responsive Top Fixed 2' },
	{ href: '/responsive/top-fix-3', name: 'Responsive Top Fixed 3' }
];

export const menuListResponsiveTransition: MenuType[] = [
	{ href: '/responsive-transition/blur', name: 'Responsive Blur' },
	{ href: '/responsive-transition/fade', name: 'Responsive Fade' },
	{ href: '/responsive-transition/fly-x', name: 'Responsive Fly-x' },
	{ href: '/responsive-transition/fly-y', name: 'Responsive Fly-y' },
	{ href: '/responsive-transition/slide', name: 'Responsive Slide' }
];

export const menuListResponsiveNoTransition: MenuType[] = [
	{ href: '/responsive-no-transition/top-fix', name: 'Responsive No Transition' },
	{ href: '/responsive-no-transition/top-fix-2', name: 'Responsive No Transition 2' },
	{ href: '/responsive-no-transition/top-fix-3', name: 'Responsive No Transition 3' }
];

export const menuListTransition: MenuType[] = [
	{ href: '/transitions/slide', name: 'Slide' },
	{ href: '/transitions/fade', name: 'Fade' },
	{ href: '/transitions/fly-x', name: 'Fly X' },
	{ href: '/transitions/fly-y', name: 'Fly Y' },
	{ href: '/transitions/blur', name: 'Blur' }
];

export const menuList: MenuType[] = [
	{ href: '/fixed-menu', rel: 'external', name: 'Fixed Top Menu' },
	{ href: '/inert', rel: 'external', name: 'Inert' },
	{ href: '/multi-custom-style', rel: 'external', name: 'Multiple-menu Custom Style' },
	{ href: '/multi-default', rel: 'external', name: 'Multiple-menu Default' },
	{ href: '/scroll-sidebar', rel: 'external', name: 'Scroll Sidebar' },
	{ href: '/sidebar-custom-style', rel: 'external', name: 'Sidebar Custom Style' },
	{ href: '/topmenu', rel: 'external', name: 'Non-fixed Top Menu' }
];

export const allMenuList: MenuType[] = [
	{ href: '/', name: 'Sidebar Default' },
	{ href: '/responsive/top-fix', name: 'Responsive Top Fixed' },
	{ href: '/responsive/top-fix-2', name: 'Responsive Top Fixed 2' },
	{ href: '/responsive/top-fix-3', name: 'Responsive Top Fixed 3' },
	{ href: '/responsive-transition/blur', name: 'Responsive blur' },
	{ href: '/responsive-transition/fade', name: 'Responsive fade' },
	{ href: '/responsive-transition/fly-x', name: 'Responsive fly-x' },
	{ href: '/responsive-transition/fly-y', name: 'Responsive fly-y' },
	{ href: '/responsive-transition/slide', name: 'Responsive slide' },
	{ href: '/transitions/slide', name: 'Slide' },
	{ href: '/transitions/fade', name: 'Fade' },
	{ href: '/transitions/fly-x', name: 'Fly X' },
	{ href: '/transitions/fly-y', name: 'Fly Y' },
	{ href: '/transitions/blur', name: 'Blur' },
	{ href: '/responsive-no-transition/top-fix', name: 'Responsive No Transition' },
	{ href: '/responsive-no-transition/top-fix-2', name: 'Responsive No Transition 2' },
	{ href: '/responsive-no-transition/top-fix-3', name: 'Responsive No Transition 3' },
	{ href: '/fixed-menu', name: 'Fixed Top Menu' },
	{ href: '/inert', name: 'Inert' },
	{ href: '/multi-custom-style', name: 'Multiple-menu Custom Style' },
	{ href: '/multi-default', name: 'Multiple-menu Default' },
	{ href: '/scroll-sidebar', name: 'Scroll Sidebar' },
	{ href: '/sidebar-custom-style', name: 'Sidebar Custom Style' },
	{ href: '/topmenu', name: 'Top Menu' }
];

export const topMenus: TopMenuType[] = [
	{
		id: '1',
		name: 'Responsive',
		child: [
			{
				id: '2',
				name: 'Responsive Top Fixed',
				href: '/responsive/top-fix'
			},
			{
				id: '3',
				name: 'Responsive Top Fixed 2',
				href: '/responsive/top-fix-2'
			},
			{
				id: '4',
				name: 'Responsive Top Fixed 3',
				href: '/responsive/top-fix-3'
			}
		]
	},
	{
		id: '11',
		name: 'No Transition',
		child: [
			{
				id: '12',
				name: 'Responsive Top Fixed 1',
				href: '/responsive-no-transition/top-fix'
			},
			{
				id: '13',
				name: 'Responsive Top Fixed 2',
				href: '/responsive-no-transition/top-fix-2'
			},
			{
				id: '14',
				name: 'Responsive Top Fixed 3',
				href: '/responsive-no-transition/top-fix-3'
			}
		]
	},
	{
		id: '21',
		name: 'Responsive Transition',
		child: [
			{
				id: '22',
				name: 'Blur Transition',
				href: '/responsive-transition/blur'
			},
			{
				id: '23',
				name: 'Fade Transition',
				href: '/responsive-transition/fade'
			},
			{
				id: '24',
				name: 'Fly X Transition',
				href: '/responsive-transition/fly-x'
			},
			{
				id: '25',
				name: 'Fly Y Transition',
				href: '/responsive-transition/fly-y'
			},
			{
				id: '26',
				name: 'Slide Transition',
				href: '/responsive-transition/slide'
			}
		]
	},
	{
		id: '31',
		name: 'Others',
		child: [
			{
				id: '32',
				name: 'Fixed Top Menu',
				href: '/fixed-menu'
			},
			{
				id: '33',
				name: 'Inert',
				href: '/inert'
			},
			{
				id: '34',
				name: 'Multiple-menu Custom Style',
				href: '/multi-custom-style'
			},
			{
				id: '35',
				name: 'Multiple-menu Default',
				href: '/multi-default'
			},
			{
				id: '36',
				name: 'Scroll Sidebar',
				href: '/scroll-sidebar'
			},
			{
				id: '37',
				name: 'Sidebar Custom Style',
				href: '/sidebar-custom-style'
			},
			{
				id: '38',
				name: 'Sidebar Top Menu',
				href: '/sidebar-topmenu'
			},
			{
				id: '39',
				name: 'Top Menu',
				href: '/topmenu'
			}
		]
	}
];

export const lorem = `Lorem ipsum dolor sit amet consectetur adipisicing elit. Soluta distinctio
autem aperiam hic quis animi magnam alias voluptas voluptatibus doloremque
nisi sapiente tempore quidem, exercitationem pariatur natus a dignissimos
consequuntur. Ea soluta voluptate voluptates dolore vel dolorum doloribus
aut pariatur. Itaque, quam molestiae adipisci vel libero aspernatur, ea quae
voluptatibus, ullam dignissimos iusto doloremque voluptatem saepe iste
expedita perferendis quasi! Atque necessitatibus vel distinctio, voluptatum
culpa tempora error quas illo eum impedit. Necessitatibus vitae eos tenetur,
ipsa, modi odio architecto rerum ipsum incidunt eligendi asperiores.
Deserunt saepe dolor neque dolorum! Animi eveniet cupiditate cum sint, quam,
praesentium illum optio ipsum neque deserunt magni, sed vitae minima eius
quis laudantium quisquam accusamus sapiente necessitatibus et quidem! Ipsum
inventore nemo nam deserunt. Beatae deleniti quod aspernatur labore natus
iusto odit, quaerat modi culpa repudiandae, voluptatum eum, recusandae
voluptas corrupti cum soluta quos facere a in animi laudantium vel esse
veniam quibusdam? Quis.

Lorem ipsum dolor sit amet consectetur adipisicing elit. Soluta distinctio
autem aperiam hic quis animi magnam alias voluptas voluptatibus doloremque
ipsa, iusto odit, quaerat modi culpa repudiandae, voluptatum eum, recusandae
voluptas corrupti cum soluta quos facere a in animi laudantium vel esse
veniam quibusdam? Quis.`;
